cloudinit/config/cc_lxd.py

```diff
@@ -210,6 +210,7 @@ def handle(name: str, cfg: Config, cloud: Cloud, args: list) -> None:
             f" '{type(lxd_cfg).__name__}'"
         )
 
+    cloud.run("snap-seeded", util.wait_for_snap_seeded, [], freq=PER_ONCE)
```

Comment:
I don't think that this is in use yet in any of our configuration modules, which I think is intentional, since the modules themselves are run with a configurable frequency. I'm curious about the benefits / costs of using this interface. 

### Specific to this use case

Are we sure that `PER_ONCE` works in all cases? Is forcing snap to un-seed possible? If so, I'd lean towards running `wait_for_snap_seeded()` unconditionally in these three modules.

Is using a fs-based semaphore cheaper than unconditionally using `snap wait system seed.loaded`? The command runs in <0.05 for me.

### More generally about `cloud.run()`

This interface has overlapping behavior and intent with `lru_cache`. `lru_cache` should only persist across stages when pickled, and I see that this interface also has the advantage of flexibility over `lru_cache()` for aggregating multiple callsites together under a single semaphore.

Do you think that we should we use `cloud.run()` over `lru_cache` in other cases as well, to avoid per-stage calls?

In this section of my comment, the first paragraph and answers to questions in the second would be well suited for the docstring of `cloud.run()`. This would make reviewing and maintenance of this code to be easier.

Reply:
lru_cache won't persist even in our pickled objects such as our Datasource. If I add decorate a datasource method as lru_cached and call it per boot stage `.cache_info()` still shows it as empty every single boot stage even with deserialization.
`CacheInfo(hits=0, misses=0, maxsize=128, currsize=0)`.

We'd have to setup a file-based memory backend I think (maybe `joblib.Memory`?) to persist and reference that cache across boot stages to load and reference previously cached method/function calls.

So I think our `cloud.run()` semaphore file is the simplest way we can persist  this cached state across our separate python invocations in each boot stage.

1. PER_ONCE directive will work in all cases because we can't un-seed snaps. seeded snaps could be removed after seeding takes place with external snap commands, but I think worrying about that is off-path/unsupported by cloud-init.
2. the `lru_cache` will only apply once a given function has been memoized. 
3. Is fs-based semaphore cheaper than subp `snap wait system seed.loaded`? Yes much cheaper
Here is a timeit run of 1000 attempts for cloud.run vs subp shelling out to call snap wait system seed.loaded:
```
root@nnn:~# python3 1.py 
use cloud.run semaphore
0.0023272480175364763
invoke subp.subp(snap wait system seed.loaded
0.5457840269955341
```

I'll add a docstr to cloud.run to better explain the use-cases for this method.